### PR TITLE
Seq-cutoff promotion count secuity tests

### PR DESCRIPTION
**Pull request by mmichal10**
Signed-off-by: Michal Mielewczyk <michal.mielewczyk@intel.com>

### CODE REVIEW

**Review by jfckm — DISMISSED** [review]

**Review by jfckm — APPROVED** [review]

File: tests/functional/pyocf/types/core.py [hunk 1]
```diff
@@ -178,6 +178,18 @@ def set_seq_cut_off_threshold(self, threshold):
...
+        self.cache.write_lock()
+
+        status = self.cache.owner.lib.ocf_mngt_core_set_seq_cutoff_promotion_count(
+            self.handle, count
+        )
+        if status:
+            self.cache.write_unlock()
+            raise OcfError("Error setting core seq cut off policy promotion threshold", status)
+
+        self.cache.write_unlock()
```
Comment (rafalste) [inline tests/functional/pyocf/types/core.py:191]: Maybe put `self.cache.write_unlock()` before checking status? Just like in the previous function.

File: tests/functional/tests/security/test_management_fuzzy.py [hunk 2]
```diff
@@ -132,6 +132,75 @@ def test_neg_cache_set_seq_cut_off_policy(pyocf_ctx, cm, cls):
...
+    cache = Cache.start_on_device(cache_device, cache_mode=cm, cache_line_size=cls)
+
+    # Create core device
+    core_device1 = Volume(S.from_MiB(10))
+    core1 = Core.using_device(core_device1, name="core1")
+
+    # Add cores
+    cache.add_core(core1)
+
+    # Change cache seq cut off promotion count to invalid one and check if failed
```
Comment (rafalste) [inline tests/functional/tests/security/test_management_fuzzy.py:193]: core instead of cache

File: tests/functional/pyocf/types/core.py [hunk 3]
```diff
@@ -160,23 +160,29 @@ def set_seq_cut_off_policy(self, policy: SeqCutOffPolicy):
...
 
+    def set_seq_cut_off_promotion(self, count):
+        self.cache.write_lock()
+
+        status = self.cache.owner.lib.ocf_mngt_core_set_seq_cutoff_promotion_count(
+            self.handle, count
+        )
         self.cache.write_unlock()
+        if status:
+            raise OcfError("Error setting core seq cut off policy promotion threshold", status)
```
Comment (rafalste) [inline tests/functional/pyocf/types/core.py:185]: promotion count?

File: tests/functional/tests/security/test_management_fuzzy.py [hunk 4]
```diff
@@ -132,6 +137,75 @@ def test_neg_cache_set_seq_cut_off_policy(pyocf_ctx, cm, cls):
...
+
+    # Add core
+    cache.add_core(core1)
+
+    # Change core seq cut off promotion count to invalid one and check if failed
+    for i in RandomGenerator(DefaultRanges.UINT32):
+        if i in ConfValidValues.seq_cutoff_promotion_range:
+            continue
+        with pytest.raises(
+            OcfError, match="Error setting core seq cut off policy promotion threshold"
```
Comment (rafalste) [inline tests/functional/tests/security/test_management_fuzzy.py:203]: promotion count?

**Review by jfckm — APPROVED** [review]

**Review by rafalste — APPROVED** [review]

**Pull request merged**